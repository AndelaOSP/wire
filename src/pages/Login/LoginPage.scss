@import "../../theme";

.login-page {
  display: flex;
  flex-direction: row;
  height: 100vh;

  .left-column {
    background-color: map_get($grey, 200);
    width: 25vw;

    .logo-image {
      margin: 1.8rem;
      height: 2.5rem;
      width: 4.5rem;
    }
  }

  .right-column {
    flex: 1;

    .login-container {
      display: flex;
      flex-direction: column;
      align-items: center;
      height: 100vh;
      justify-content: center;
      margin: auto !important;

      .title {
        font-size: 1.2rem;
        margin-bottom: 0.3rem;
      }

      .button {
        margin: 2rem;
        width: 10rem;
        padding: 1.2rem;
        font-size: small;
      }
    }

    .google-logo {
      width: 1.5rem;
      height: 1.5rem;
    }
  }
}
