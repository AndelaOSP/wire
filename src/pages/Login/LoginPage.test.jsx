
import React from 'react';
import { shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';
import { LoginPage } from './LoginPage.Component';
import authenticateUser from '../../helpers/auth';

jest.mock('../../helpers/auth');

authenticateUser.isAuthenticated = false;

const props = {
  location: {
    state: {
      from: {},
    },
  },
  isError: true,
  getToken: jest.fn(),
  hasToken: true,
};

describe('LoginPage component', () => {
  let loginPage;
  let instance;

  beforeEach(() => {
    loginPage = shallow(<LoginPage {...props} />);
    instance = loginPage.instance();
  });

  it('should render without crashing', () => {
    
    const raisedButton = loginPage.find('.button');

    expect(raisedButton.length).toEqual(1);
    expect(instance.props.isError).toEqual(true);
    expect(instance.props.hasToken).toEqual(true);
    expect(shallowToJSON(loginPage)).toMatchSnapshot();
  });

  it('should not login when token is null', () => {    
    loginPage.setProps({
      hasToken: false,
    });
    const textToFind = 'Welcome to Wire Please sign in with your Google account to proceed';
    expect(loginPage.find('.welcome-text').text()).toEqual(textToFind);
  });
});
