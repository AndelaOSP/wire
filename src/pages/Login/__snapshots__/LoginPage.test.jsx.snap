// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component should render without crashing 1`] = `
<div>
  <div
    className="login-page"
  >
    <div
      className="left-container"
    >
      <div>
        <img
          className="andela-logo"
          src="/assets/images/andelaLogo.png"
        />
      </div>
      <div
        className="welcome-text"
      >
        <p>
          Welcome to
           
          <span
            className="wire"
          >
            Wire 
          </span>
          <br />
          Please sign in with your Google account to proceed
        </p>
      </div>
      <RaisedButton
        className="button"
        disabled={false}
        fullWidth={false}
        href="undefined/login?redirect_url=undefined/login"
        icon={
          <img
            className="google-logo"
            src="../../../assets/images/icons8-google.svg"
          />
        }
        label={
          <p
            className="label"
          >
            Sign In With Google
          </p>
        }
        labelPosition="after"
        primary={false}
        secondary={false}
        style={
          Object {
            "height": "3rem",
            "marginLeft": "2vw",
            "position": "relative",
            "width": "15rem",
          }
        }
      />
    </div>
    <div
      className="right-container"
    >
      <img
        className="landing-image"
        src="/assets/images/wire_landing_page_vector@2x.png"
      />
      <div
        className="right-text"
      >
        <p>
          <span
            style={
              Object {
                "fontWeight": 600,
              }
            }
          >
            An Incident
          </span>
           
          <br />
          Reporting Platform
        </p>
      </div>
      <div
        className="underline"
      />
    </div>
  </div>
  <CustomNotification
    autoHideDuration={150000}
    message=""
    open={true}
    type="error"
  />
</div>
`;
